pkg/controller/daemon/daemon_controller.go

```diff
@@ -865,12 +884,29 @@ func (dsc *DaemonSetsController) podsShouldBeOnNode(
 				continue
 			}
 			if pod.Status.Phase == v1.PodFailed {
+				failedPodsObserved++
+
+				// This is a critical place where DS is often fighting with kubelet that rejects pods.
+				// We need to avoid hot looping and backoff.
+				backoffKey := failedPodsBackoffKey(ds, node.Name)
+				delay := dsc.failedPodsBackoff.Get(backoffKey)
```

Comment:
move `delay` inside the `if inBackoff {` block?

Reply:
fixed, thx